client/ui/channelsresolver.go

```diff
@@ -0,0 +1,137 @@
+package ui
+
+import (
+	"encoding/json"
+
+	"github.com/8tomat8/SSU-Golang-252-Chat/client/config"
+	"github.com/8tomat8/SSU-Golang-252-Chat/client/contacts"
+	"github.com/8tomat8/SSU-Golang-252-Chat/client/listener"
+	"github.com/8tomat8/SSU-Golang-252-Chat/loger"
+	"github.com/8tomat8/SSU-Golang-252-Chat/messageService"
+	"github.com/8tomat8/SSU-Golang-252-Chat/userinfo"
+)
+
+func channelsResolver() {
+	for {
+		select {
+		case <-listener.QuitChannel:
+			return
+		case msg := <-listener.AuthorizationChannel:
+			receiveAuth(msg)
+		case msg := <-listener.MessageChannel:
+			receiveMessage(msg)
+		case msg := <-listener.SettingsChannel:
+			receiveSettings(msg)
+		case <-listener.ContactsChannel:
+		}
+	}
+}
+
+func receiveAuth(message messageService.Message) {
+	switch message.Header.Command {
+	case config.GetConfig().MessageCommand.LoginIsSucc:
+		//login data is valid, so we open other channels to listen to server
+		listener.MessageChannel = make(chan messageService.Message)
+		listener.SettingsChannel = make(chan messageService.Message)
+		listener.ContactsChannel = make(chan messageService.Message)
+		//save received username and token
+		userinfo.CurrentUserInfo.UserName = message.Header.UserName
+		userinfo.CurrentUserInfo.UserName = message.Header.Token
+		//now we just inform UI that login was successfully
+		loger.Log.Infof("User %s login successfully.", message.Header.UserName)
+		qmlLogin.LoginDataIsValid(true)
+		qmlStatus.SendStatus("Loging successfully. Be happy and write messages :)")
+	case config.GetConfig().MessageCommand.LoginIsNotSucc:
+		//login data is invalid, so we close conn and inform UI
+		listener.QuitChannel <- struct{}{}
+		//we blocked Auth channel until user want to send login data again
+		listener.AuthorizationChannel = nil
+		//just close connection
+		err := connection.Close()
+		if err != nil {
+			loger.Log.Errorf("Cannot close connection after login is not successfully. %s", err)
+			//for client it doesn`t matter. connection can`t close == login is not successfully
+		}
+		loger.Log.Infof("User %s login wasn't successfully.", message.Header.UserName)
+		qmlLogin.LoginDataIsValid(false)
+		qmlStatus.SendStatus("Loging was unsuccessfully. Please check username or password again.")
+	case config.GetConfig().MessageCommand.RegisterIsSucc:
+		//however register successfully we need to close connection
+		//because all register actions have ended
+		listener.QuitChannel <- struct{}{}
+		//we blocked Auth channel until user want to send Auth data again
+		listener.AuthorizationChannel = nil
+		err := connection.Close()
+		if err != nil {
+			loger.Log.Errorf("Cannot close connection after register is not successfully. %s", err)
+			qmlRegister.RegisterDataIsValid(false)
```

Comment:
We should do nothing, when error happend, because coonection may be closed in channelsresolver function. So error spawn, but it's has no effect on register.
See previous error check